draft-irtf-cfrg-vdaf.md

```diff
@@ -4004,39 +4004,24 @@ The next variant of Prio3 supports summing of integers in a pre-determined
 range. Each measurement is an integer in the range `[0, max_measurement]`,
 where `max_measurement` defines the largest valid measurement.
 
-The range check is accomplished by encoding the measurement as a bit vector,
-encoding the measurement plus an offset as a bit vector, then checking that the
-two encoded integers are consistent. Let
-
-* `bits = max_measurement.bit_length()`, the number of bits needed to encode
-  the largest valid measurement
-
-* `offset = 2 ** bits - 1 - max_measurement`
-
-The first bit-encoded integer is the measurement itself. Note that only
-measurements between `0` and `2**bits - 1` can be encoded this way with as many
-bits. The second bit-encoded integer is the sum of the measurement and
-`offset`. Observe that this sum can only be encoded this way if it is between
-`0` and `2**bits - 1`, which implies that the measurement is between `-offset`
-and `max_measurement`.
-
-The circuit first checks that each entry of both bit vectors is a one or a
-zero. It then decodes both the measurement and the offset measurement, and
-subtracts the offset from the latter. It then checks if these two values are
-equal. Since both the measurement and the measurement plus `offset` are in the
-same range of `[0, 2**bits)`, this means that the measurement itself is between
-`0` and `max_measurement`.
-
-The circuit uses the polynomial-evaluation gadget `PolyEval` specified in
+The range check is accomplished by encoding the measurement as a vector of field
+elements with value zero or one, such that a weighted sum of the "bits" can only
+be in this range. All but one of the weights are successive powers of two, as in
+a binary bit decomposition, and the last weight is chosen such that the sum of
+all weights is equal to `max_measurement`. With these weights, valid
+measurements have either one or two possible representations as a vector of 0/1
+field elements, and invalid measurements cannot be represented.
```

Comment:
"a vector of 0/1 field elements" can be consufed with vectors of length zero or one.

```suggestion
measurements have either one or two possible representations as vectors of 
field elements with zero or one values, and invalid measurements cannot be represented.
```